### PR TITLE
Add docker compose

### PR DESCRIPTION
**Pull request by Cleopha**
# Description

- Docker compose that run all backend project

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [X] I have assigned this PR to myself
- [X] I have added at least 1 reviewer
- [ ] I have tested this code
- [X] I have added sufficient documentation in the code

### CODE REVIEW

**Review by 0xpanoramix — APPROVED** [review]
LGTM !

File: .env.example [hunk 1]
```diff
@@ -13,17 +13,50 @@ REDIS_PORT=6379
...
+NOTION_CLIENT_ID=
+NOTION_CLIENT_SECRET=
+NOTION_REDIRECT_URL=http://localhost:8080/
+
+# Infura Credentials
+INFURA_ENDPOINT=
+
+# Idefix specific
+REDIS_ADDR="localhost:6379" # I guess is useless
+PRODUCER_ENDPOINT="127.0.0.1:19092" # I guess is useless
```
Comment (TomChv) [inline .env.example:54]: I agree, put it directly in compose

File: .env.example [hunk 2]
```diff
@@ -13,17 +13,50 @@ REDIS_PORT=6379
...
+TIME_INTERVAL=2
+
+# Redis configuration Idefix
+REDIS_PORT_IDEFIX=6379
+REDIS_PASS_IDEFIX=idefix_pass
+
+# Panoramix specific
+BROKER_ADDRESS=http://localhost:19092 # I guess is useless
```
Comment (TomChv) [inline .env.example:62]: Same

File: .env.example [hunk 3]
```diff
@@ -13,17 +13,50 @@ REDIS_PORT=6379
...
+INFURA_ENDPOINT=
+
+# Idefix specific
+REDIS_ADDR="localhost:6379" # I guess is useless
+PRODUCER_ENDPOINT="127.0.0.1:19092" # I guess is useless
+TIME_INTERVAL=2
+
+# Redis configuration Idefix
+REDIS_PORT_IDEFIX=6379
+REDIS_PASS_IDEFIX=idefix_pass
```
Comment (TomChv) [inline .env.example:59]: Should be empty

File: docker-compose.yml [hunk 4]
```diff
@@ -92,8 +92,96 @@ services:
     ports:
       - "${GRPC_DOC_PORT}:80"
 
+# ------------- OBELIX -------------
+
+  zookeeper:
+    image: confluentinc/cp-zookeeper:latest
+    # I have to delete the exposure of the ports if I have put in the same network ?
```
Comment (TomChv) [inline docker-compose.yml:101]: It depends if you want to expose it publicly or nope, I guess no but it's up to you

File: docker-compose.yml [hunk 5]
```diff
@@ -92,8 +92,96 @@ services:
...
+    ports:
+      - "22181:22181"
+    environment:
+      ZOOKEEPER_CLIENT_PORT: 22181
+    networks:
+      - obelix
+
+  kafka:
+    image: confluentinc/cp-kafka:latest
+    # I have to delete the exposure of the ports if I have put in the same network ?
```
Comment (TomChv) [inline docker-compose.yml:111]: It depends if you want to expose it publicly or nope, I guess no but it's up to you

**Review by TomChv — CHANGES_REQUESTED** [review]
:warning: You forget to rebase!

Looks good overall, small comments

**Review by TomChv — APPROVED** [review]
LGTM

**Pull request merged**